dataset.py

```diff
@@ -9,27 +11,43 @@
 class Dataset():
     """The Google Analytics dataset."""
 
-    def __init__(self):
+    def __init__(self, debug=False):
         """Load the data from disk."""
 
-        type_change = {"fullVisitorId": str,
-                       "sessionId": str,
-                       "visitId": str}
+        if(debug): nrows = 100
+        else:      nrows = None
+        type_change_columns = {"fullVisitorId": str,
+                        "sessionId": str,
+                        "visitId": str}
+        json_columns = ['device', 'geoNetwork', 'totals', 'trafficSource']
         self.train = pd.read_csv(os.path.join(_DATA_DIR, _TRAIN),
-                                 dtype=type_change)
-        self.test = pd.read_csv(os.path.join(_DATA_DIR, _TEST),
-                                dtype=type_change)
-
+                                 converters={column: json.loads for column in json_columns},  
+                                 dtype=type_change_columns,
+                                 nrows = nrows)
+                                
+        self.test = pd.read_csv(os.path.join(_DATA_DIR, _TEST), 
+                                converters={column: json.loads for column in json_columns},
+                                dtype=type_change_columns,
+                                nrows = nrows)
+        for column in json_columns:
+            train_column_as_df = json_normalize(self.train[column])
+            test_column_as_df = json_normalize(self.test[column])
+            self.train = self.train.drop(column, axis=1).merge(train_column_as_df, 
```

Comment:
It might be good to remove the `.drop()` to make sure that no data is accidentally deleted during this step. Generally I think the preprocessing steps should add columns, not replace them.